src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriter.java

```diff
@@ -65,13 +66,16 @@ public TempBlob removeUrlsFromIndexYamlAndWriteToTempBlob(final TempBlob index,
                                                             final Repository repository)
   {
     storageFacet = repository.facet(StorageFacet.class);
+    String remoteUrl = (String) repository.getConfiguration().attributes("proxy")
+        .get("remoteUrl");
 
-    return new StreamCopier<>(outputStream -> updateUrls(index.get(), outputStream),
+    return new StreamCopier<>(outputStream -> updateUrls(index.get(), outputStream, remoteUrl),
```

Comment:
Won't this return a URL to the remote to the client therefore causing the client to circumvent NXRM?

Reply:
@sergeyonofreychuck can your team double check this?

Reply:
I tested it on a small environment without having issues. Can you please merge it?